### ISSUE TITLE
svg tag in template incorrectly marked red when closing tag spans more than one line

### ISSUE THREAD

**Issue by zmwangx** [johnsoncodehk/volar#192]
Consider this simple example

```vue
<template>
  (<svg height="100" width="100">
    <circle cx="50" cy="50" r="40" stroke="black" stroke-width="3" fill="red" /></svg
  >)
  <p></p>
</template>
```

produced by Prettier. It's not a shining example of good formatting, but it's valid HTML and valid Vue template. (I can share the Prettier config, but that's not terribly relevant.) Volar doesn't like the svg closing tag though, and marks it and everything after it in red:

![image](https://user-images.githubusercontent.com/4149852/118848118-09799180-b901-11eb-95d4-4699e5a60223.png)

I only noticed this problem with svg. I wonder what's causing it.

## Version info

Volar v0.25.8
VSCode:
```
Version: 1.56.2
Commit: 054a9295330880ed74ceaedda236253b4f39a335
Date: 2021-05-12T17:44:30.902Z
Electron: 12.0.4
Chrome: 89.0.4389.114
Node.js: 14.16.0
V8: 8.9.255.24-electron.0
OS: Darwin x64 20.4.0
```

**Comment by johnsoncodehk** [comment]
Thanks for you report! This is html syntax highlighting bug from vscode, svg end tag broken when new line.

** this picture is html document not vue document.
<img width="331" alt="螢幕截圖 2021-05-20 上午12 38 18" src="https://user-images.githubusercontent.com/16279759/118850776-b1905a00-b903-11eb-94b7-dccba6b1f2b2.png">

**Comment by zmwangx** [comment]
Yeah sorry, didn't occur to me that I should check html first... I'll report upstream.

**Comment by zmwangx** [comment]
https://github.com/microsoft/vscode/issues/124215

**Comment by johnsoncodehk** [comment]
Should fixed by https://github.com/microsoft/vscode/pull/124220, embedded languages syntax highlight updated in 0.25.4, so you could use volar 0.25.3 for now before vscode fixed it.

**Comment by zmwangx** [comment]
Thanks for looking into the upstream issue! Unfortunately they delegate it to textmate/html.tmbundle which seems to hardly get any attention, and the issue has been outstanding for years: https://github.com/textmate/html.tmbundle/issues/101 I wonder if it's even worth the effort to submit your PR there instead.

**Comment by zmwangx** [comment]
Somewhat related (but probably not the same bug): I noticed another syntax highlighting problem since v0.25.4:

```vue
<template>
  <div>
    <div>
      <div>
        <template
          v-for="index in 10"
          :key="index"
        ></template>
      </div>
    </div>
  </div>
</template>
```

![image](https://user-images.githubusercontent.com/4149852/118981803-561aa680-b9ad-11eb-84e3-29dff8bd0bd4.png)

At a certain point tags become plaintext.

This piece of code actually highlights alright in HTML mode.

Should I open a new issue?

**Comment by johnsoncodehk** [comment]
> Thanks for looking into the upstream issue! Unfortunately they delegate it to textmate/html.tmbundle which seems to hardly get any attention, and the issue has been outstanding for years: [textmate/html.tmbundle#101](https://github.com/textmate/html.tmbundle/issues/101) I wonder if it's even worth the effort to submit your PR there instead.

I am not familiar with this project, and I don't have time to check it at the moment.

> Somewhat related (but probably not the same bug): I noticed another syntax highlighting problem since v0.25.4:

Will fix in next version, thanks to report!

> Should I open a new issue?

No need because I already got this problem, but if you want to track this problem status you can open a new issue, thanks. :)

**Comment by johnsoncodehk** [comment]
`<template>` syntax highlighting problem has been fixed in 0.25.9.
